plone/app/linkintegrity/parser.py

```diff
@@ -30,17 +30,34 @@ def handle_starttag(self, tag, attrs):
             self.links.extend(search_attr('href', attrs))
         if tag == 'img':
             self.links.extend(search_attr('src', attrs))
+            self.links.extend(search_attr('srcset', attrs))
         if tag == 'source':
+            # Used within img/picture/audio/video tags
+            # to embed various sources of media.
             self.links.extend(search_attr('src', attrs))
             self.links.extend(search_attr('srcset', attrs))
+        if tag == 'audio':
+            # Embeds audio recordings.
+            self.links.extend(search_attr('src', attrs))
+        if tag == 'video':
+            # Embeds videos.
+            self.links.extend(search_attr('src', attrs))
+        if tag == 'iframe':
+            # Used to embed PDFs.
+            self.links.extend(search_attr('src', attrs))
 
 
 def search_attr(name, attrs):
     """Search named attribute in a list of attributes.
     """
     for attr, value in attrs:
         if attr == name:
-            return [value]
+            if name == "srcset":
+                # SRCSET is split by commas, and each line's first
+                # element is the URL in question.
+                return [x.strip().split()[0] for x in value.split(",")]
```

Comment:
what happen if value is empty or malformed? Can happen?

Reply:
https://docs.python.org/3/library/html.parser.html the documentation is unclear, other than saying explicitly that the value is always present (I believe empty tag attribute values are equivalent to the tag attribute being absent).